File: PE-common/src/main/java/nl/adaptivity/ws/soap/SoapSeeAlso.java
package nl.adaptivity.ws.soap;

import org.jetbrains.annotations.NotNull;

import static java.lang.annotation.ElementType.METHOD;
import static java.lang.annotation.ElementType.PARAMETER;
import static java.lang.annotation.ElementType.TYPE;
import static java.lang.annotation.RetentionPolicy.RUNTIME;

import java.lang.annotation.Retention;
import java.lang.annotation.Target;

/**
 * Like @XmlSeeAlso, but allowed on more types.
 * @author pdvrieze
 *
 */
@Target(value={TYPE, METHOD, PARAMETER})
@Retention(RUNTIME)
public @interface SoapSeeAlso {
  @NotNull Class<?>[] value();
}

File: PE-common/src/jvmMain/kotlin/nl/adaptivity/ws/soap/SoapSeeAlso.kt
/*
 * Copyright (c) 2018.
 *
 * This file is part of ProcessManager.
 *
 * ProcessManager is free software: you can redistribute it and/or modify it under the terms of version 3 of the
 * GNU Lesser General Public License as published by the Free Software Foundation.
 *
 * ProcessManager is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even
 * the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License along with ProcessManager.  If not,
 * see <http://www.gnu.org/licenses/>.
 */

package nl.adaptivity.ws.soap

import kotlin.reflect.KClass

/**
 * Like @XmlSeeAlso, but allowed on more types.
 * @author pdvrieze
 */
@Target(AnnotationTarget.CLASS, AnnotationTarget.FILE, AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER,
        AnnotationTarget.PROPERTY_SETTER, AnnotationTarget.VALUE_PARAMETER)
@Retention(AnnotationRetention.RUNTIME)
annotation class SoapSeeAlso(vararg val value: KClass<*>)
